File: AlSpark/src/main/scala/com/al/entity/Log.scala
package com.al.entity

import com.al.util.StringUtil

import scala.beans.BeanProperty

/**
  * Created by An on 2016/11/24.
  */
class Log {
  @BeanProperty var Ts: Long = 0L
  @BeanProperty var Ip: String = ""
  @BeanProperty var Uuid: String = ""

  @BeanProperty var SearchEngine: String = ""
  @BeanProperty var Country: String = ""
  @BeanProperty var Area: String = ""

  @BeanProperty var ContentId: Long = 0L
  @BeanProperty var Url: String = ""
  @BeanProperty var Title: String = ""

  @BeanProperty var wd: Wd = null

  def getPagetype(): Char = {
    if(wd != null && wd.getT() != null) {
      val wdt: String  = wd.getT()
      if(wdt.length() == 3) {
        val pt: Char = wdt.charAt(2)
        if(pt >= '0' && pt <= '5') {
          return pt
        }
      }
    }
    return '6'
  }

  def getClearTitle(): String = {
    var title = ""
    if(Title != null) {
      title = StringUtil.clearTitleAll(Title)
    }
    return title
  }

  /**
    * 日志是否合法
    * @return
    */
  def isLegal(): Boolean = {
    if(Ip == null || Ip.equals("") || Uuid == null || Uuid.equals("")) {
      return false
    }
    return true
  }
}

class Wd {
  @BeanProperty var t: String = ""
}

File: AlHadoop/src/main/java/com/al/entity/Log.java
package com.al.entity;

import com.al.util.StringUtil;

public class Log {
	private String Act;
	private long Ts;
	private String Ip;
	private String Uuid;
	private String Uid;
	private String ContentId;
	private String Url;
	private String Title;
	private String Refer;
	private String Ua;
	private String Country;
	private String Area;
	private Wd wd;
	private char pagetype;

	/**
	 * 日志是否合法
	 * @return
	 */
	public boolean isLegal() {
		if(Ip == null || Ip.equals("") || Uuid == null || Uuid.equals("")) {
			return false;
		}
		return true;
	}
	
	public char getPagetype() {
		Wd wd = getWd();
		if(wd != null && wd.getT() != null) {
			String wdt = wd.getT();
			if(wdt.length() == 3) {
				char pt = wdt.charAt(2);
				if(pt >= '0' && pt <= '5') {
					pagetype = pt;
				}
			}
		}
		return pagetype;
	}

	public String getClearTitle() {
		String title = "";
		if(Title != null) {
			title = StringUtil.clearTitleAll(Title);
		}
		return title;
	}
	
	class Wd {
		String t;

		public String getT() {
			return t;
		}
		public void setT(String t) {
			this.t = t;
		}
	}
	
	public String getAct() {
		return Act;
	}
	public void setAct(String act) {
		Act = act;
	}
	public long getTs() {
		return Ts;
	}
	public void setTs(long ts) {
		Ts = ts;
	}
	public String getIp() {
		return Ip;
	}
	public void setIp(String ip) {
		Ip = ip;
	}
	public String getUuid() {
		return Uuid;
	}
	public void setUuid(String uuid) {
		Uuid = uuid;
	}
	public String getUid() {
		return Uid;
	}
	public void setUid(String uid) {
		Uid = uid;
	}
	public String getContentId() {
		return ContentId;
	}
	public void setContentId(String contentId) {
		ContentId = contentId;
	}
	public String getUrl() {
		return Url;
	}
	public void setUrl(String url) {
		Url = url;
	}
	public String getTitle() {
		return Title;
	}
	public void setTitle(String title) {
		Title = title;
	}
	public String getRefer() {
		return Refer;
	}
	public void setRefer(String refer) {
		Refer = refer;
	}
	public String getUa() {
		return Ua;
	}
	public void setUa(String ua) {
		Ua = ua;
	}
	public String getCountry() {
		return Country;
	}
	public void setCountry(String country) {
		Country = country;
	}
	public String getArea() {
		return Area;
	}
	public void setArea(String area) {
		Area = area;
	}
	public Wd getWd() {
		return wd;
	}
	public void setWd(Wd wd) {
		this.wd = wd;
	}
	public void setPagetype(char pagetype) {
		this.pagetype = pagetype;
	}
	
}
